contribute/design/1.x/ndm/20190705-ndm-disk-identification.md

```diff
@@ -0,0 +1,237 @@
++------------------+        +-----------+      +-----------+      +-------------------+
+     |                        |                     |                  |
+     |No                      |                     |                  |
+     v                        |Yes                  |Yes               |
++-----------+                 |                     |                  |
+|if DEVTYPE |                 |                     |                  |
+| equals    |                 |                     |                  |
+|"partition"|                 |                     |                  |
+|add part   |                 |                     |                  |
+|number to  |                 |                     |                  |
+|  UID      |                 |                     |                  |
++-----------+                 |                     |                  |
+     |                        |                     |                  |
+     |                        |                     |                  |
+     |                        |                     |                  |
++----v------+                 |                     |                  |
+|           |                 |                     |                  |
+|Use md5 of |                 |                     |                  |
+| generated |                 |                     |                  |
+|   UID     |<----------------+---------------------+-------------------
+|           |
+|           |
++-----------+
+
+
+```
+
+## Graduation Criteria
+
+- NDM should be able to uniquely identify the disks, across reboots, across different
+  SCSI ports and anywhere within the cluster. NDM should be also able to create
+  N partition as per the user config. The unique identification can be marked
+  successful, if NDM detects the disk, and the user is able to retrieve his data
+  from the disk
+
+## Implementation History
+
+- Owner acceptance of `Summary` and `Motivation` sections - YYYYMMDD
+- Agreement on `Proposal` section - YYYYMMDD
+- Date implementation started - YYYYMMDD
+- First OpenEBS release where an initial version of this OEP was available - YYYYMMDD
+- Version of OpenEBS where this OEP graduated to general availability - YYYYMMDD
+- If this OEP was retired or superseded - YYYYMMDD
+
+## Drawbacks
+
+- Do not have a mechanism to identify disks that are attached in multipath mode.
+- Cannot identify disks and create BDs if the same disk is connected to 2 nodes 
+  at the same time.
+- The partition created on the disk for identification will remain even after NDM
+  is uninstalled from the node.
+- Consumers of NDM will not be able to create a partition on blockdevices on which
+  NDM already created a partition for identification. 
+
+## Infrastructure Needed
+
+- Test setup with different types of disks in different configurations to test and
+  ensure the unique ID generation process.
```

Comment:
The implementation looks good to proceed. As the implementation progresses, few negative scenarios need to be addressed before graduating this feature:
- What happens if the partition or filesystem is deleted? Even though this is not expected of the user, this kind of case should be easy to detect. 
- The upgrade strategy from older version device ids to the new IDs needs to be outlined.

Reply:
Done.